tools/bl.c

```diff
@@ -378,45 +378,34 @@ static int bl_cmd_channel_conf(int argc, char *argv[])
 		return EXIT_FAILURE;
 	}
 
-	if (!read_sized_uint(argv[ARG_CHANNEL], &channel, sizeof(msg.channel_conf.channel))) {
-		fprintf(stderr, "Failed to parse value.\n");
-		return EXIT_FAILURE;
-	}
-	msg.channel_conf.channel = channel;
-
-	if (!read_sized_uint(argv[ARG_GAIN], &gain, sizeof(msg.channel_conf.gain))) {
-		fprintf(stderr, "Failed to parse value.\n");
-		return EXIT_FAILURE;
-	}
-	msg.channel_conf.gain = gain;
+	success = true;
+	success &= read_sized_uint(argv[ARG_CHANNEL], &channel, sizeof(msg.channel_conf.channel));
```

Comment:
I'd always do "success = success & <expr>" because &= is bitwise rather than logical. In this case it should always work because the bools should always be 0 or 1, but bool is actually a uint8_t (or larger) and can hold true values which aren't 1.

Reply:
I guess you meant "success = success && " for logical AND, but yeah, I has happy to use bitwise AND for this case.